### DISCUSSION TITLE
Idiomatic Gitops way to upgrade fluxcd?

### DISCUSSION THREAD

**Discussion by pdeva** [fluxcd/flux2#2926] [Q&A] [answered]
From what I have read, seem the only way to upgrade fluxcd installation on a cluster is to rerun the `flux bootstrap` command. This however requires one to be running on a machine with access to the k8s cluster which can be a pain for clusters in production environment.

Is there a (recommended) Gitops way to upgrade fluxcd on a cluster?

**Comment by stefanprodan** [comment, marked as answer]
The recommend way is to let Flux upgrade itself by running `flux install --export` with the same args as bootstrap and commit the changes to the repo. We offer a [GitHub Action](https://github.com/fluxcd/flux2/tree/main/action#automate-flux-updates) for automating the whole process.

**Comment by masterkain** [comment]
given a bootstrap like this:

```
flux bootstrap github --context=mycontext --owner=myowner --repository=myclusterrepo --path=clusters/mytenant --branch=main --read-write-key --components-extra=image-reflector-controller,image-automation-controller
```

is the following enough in the github update action?

```
flux install --components-extra="image-reflector-controller,image-automation-controller" --export > ./clusters/mytenant/flux-system/gotk-components.yaml
```

**Reply by stefanprodan** [reply]
Yes
